src/infra/ConversionTrackingReporter.ts

```diff
@@ -0,0 +1,75 @@
+import { ConversionEvent, COOKIE_PARAM, ListingsClickEvent } from '../models';
+import { DEFAULT_CONVERSION_TRACKING_DOMAIN, LISTINGS_SOURCE_PARAM } from '../models/constants';
+import { CommonConversionData } from '../models/conversiontracking/CommonConversionData';
+import { ConversionTrackingService, HttpRequesterService } from '../services';
+import { calculateSeed } from './CalculateSeed';
+
+const conversionEndpoint = 'conversiontracking/conversion';
+const listingsEndpoint = 'listings';
+
+export class ConversionTrackingReporter implements ConversionTrackingService {
+  private _debug: boolean|undefined;
+  constructor(
+    private httpRequesterService: HttpRequesterService,
+    private debug?: boolean|undefined,
+  ) {
+    this._debug = debug;
+  }
+
+  /**
+   * Prints event details to the console for debugging of analytics events as they fire.
+   * @param event - the details of the event that will be printed
+   * @param type - the type of the event that was tracked
+   */
+  private printEvent(event: string, type: string): void {
+    if (!this._debug) return;
+    console.log(
+      `%c[YextAnalytics]%c- Tracked ${type} event: ${event}`,
+      'background: white; color: blue;',
+      '',
+    );
+  }
+
+  async handleRequest(url: string): Promise<void> {
+    const res = await this.httpRequesterService.get(url.toString());
+    if (res.status !== 200 && !(res.type === 'opaqueredirect' || res.type === 'opaque')) {
+      const errorMessage = await res.text();
+      throw new Error(errorMessage);
+    }
+  }
+
+  private static formatBaseEvent(event: CommonConversionData, params: URLSearchParams): void {
+    params.set(COOKIE_PARAM, event.cookieId);
+    if (event.referrer) params.set('referrer', event.referrer);
+    params.set('v', calculateSeed().toString());
+  }
+
+  /** {@inheritDoc ConversionTrackingService.trackConversion} */
+  async trackConversion(event: ConversionEvent): Promise<void> {
```

Comment:
Note we also need to pass a location field for the conversion event as well(its should be the same as the listings location param(currently for ytag it comes from window.location.href)

Reply:
I didn't see that on the required param for the conversion event itself, just the listingsclickevent in the docs, it's not even listed as an optional parameter on the conversion event: https://www.yext.com/docmd/docs/src/com/yext/analytics/events/conversions.html

Reply:
Ah yea location is a new param we added in the last few months for conversions(its not required for the event to work though). Its mainly to support our new analytics events pipeline. I'll update those docs.